availability-store/src/lib.rs

```diff
@@ -119,10 +129,17 @@ impl Store {
 		);
 
 		if let Some(extrinsic) = data.extrinsic {
+			let encoded = extrinsic.inner.encode();
 			tx.put_vec(
 				columns::DATA,
 				extrinsic_key(&data.relay_parent, &data.candidate_hash).as_slice(),
-				extrinsic.encode(),
+				encoded.clone(),
+			);
+
+			tx.put_vec(
```

Comment:
So this is storing the message data ready for some later recall under some circumstances? If those circumstances are not already documented, maybe put them here. If they are then a pointer to them somewhere near here would be most useful for anyone coming to this code.